### ISSUE TITLE
make  detect_language return Option<Lang>

### ISSUE THREAD

**Issue by gengjun** [quickwit-oss/whichlang#16]
Currently, this function can only return one of 16 supported languages. what if the input text is not one of them ?

````rust
pub fn detect_language(text: &str) -> Lang;
````

**Comment by sagarishere** [comment]
I agree with this. It should be very clear that the language has not been detected, and English should not be default. 

Let the user decide which one is the default language they want to fall back upon, we can provide them with knowledge that the language could not be detected. 

Eng as a default fallback is making it difficult to use for me.
